### PR TITLE
Fix Popover Initialization

### PR DESCRIPTION
**Pull request by WoH**
- [x] Fix the Popover Freeze
- [x] Fix the Initial Loading of the Poke(In)Dex page
- [x] Cleanup Legacy Components
- [x] Introduce Filter Model
- [x] Propagate Changes throughout the Popover Segments
- Update the Map Page filter : **This should be done in another pr**
- Fix styling (platform specific)

### CODE REVIEW

File: ionic2/app/theme/app.md.scss [hunk 1]
```diff
@@ -35,3 +35,6 @@ navbar .searchbar-input {
   box-shadow: inset 0 0 4px 0 rgba(0,0,0,0.2);
   border: 1px solid #b1b1b1;
 }
```
Comment (AlexanderLill) [inline ionic2/app/theme/app.md.scss:37]: I would like an empty line here, but dunno if this is a good coding style from my side :)

**Review by AlexanderLill — APPROVED** [review]

File: ionic2/app/theme/app.md.scss (hunk 1 above)
Comment (WoH) [reply ionic2/app/theme/app.md.scss:37]: Done

**Comment by MajorBreakfast** [comment]
Using the controls with the mouse and touch now works perfectly. Good job

Bug: Interaction with the menu button in the navigation bar is broken. The menu won't stay open properly.

**Comment by AlexanderLill** [comment]
Bug: When the content of the shown component is longer than the height of the popover, it is extended to the bottom of the screen and further down to the floor of the room I am currently sitting in.

This can be checked on the pokemon-filter branch. How can we proceed with this? Can you maybe fix that?

**Review by AlexanderLill — APPROVED** [review]
Would be nice if the bug with "more content doesnt create a scrollbar but just keeps going downwards" could be fixed though :)

**Comment by WoH** [comment]
> Bug: Interaction with the menu button in the navigation bar is broken. The menu won't stay open properly.

Please elaborate. If the popover is open you're not supposed to be able to access the navigation menu directly. The first tap next to the popover closes it, then you can use the navbar to navigate.

> Bug: When the content of the shown component is longer than the height of the popover, it is extended to the bottom of the screen and further down to the floor of the room I am currently sitting in.
> This can be checked on the pokemon-filter branch. How can we proceed with this? Can you maybe fix that?

Explanation: To avoid issues with ionic, we render the tab view after the Popover is Loaded so the ionic event ionChange registers properly. This means ionic doesn't account for the space needed for the tab content to display on screen when it calculates the distance needed to the bottom. Therefore, we now pass the popover additional styling using the popover class that I pass to the Popover. Right now I was only considering the web view that should be similar to android and defined a global style (top: -500px). I am just trying to make the tools available for everyone else to maybe refine the styling and implement custom behavior for other platforms (iOS and wp8 seem very similar in that regard, you can look at this by going to url/ionic-lab) using the platform specific styles.

**Edit:** Added that to the todo list at the top, maybe someone can create an issue and link this wall of text.

**Comment by WoH** [comment]
Also, I noticed we should make the popover a bit wider and maybe dry it out by removing or shortening the text. As much as I enjoy reading it, it takes up too much space on a smaller screen

**Comment by WoH** [comment]
Posted the issue but I don't want to wait for the final styling. I'll merge this until we rewrite this as a simple component like @MajorBreakfast suggested.

**Pull request merged**